### PR TITLE
Update example project

### LINKED ISSUE #2636: The /example project doesn't build
Following the README, I get 

```
cabal: Could not resolve dependencies:
[__0] trying: example-0.1.0.0 (user goal)
[__1] unknown package: plutus-core (dependency of example)
[__1] fail (backjumping, conflict set: example, plutus-core)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: example, plutus-core
```

At some point `language-plutus-core` was renamed to `plutus-core`. The `example/cabal.project` file defines some github sources, which point to an older commit of this repo `73234550dc2b2c6209975789845bb5ef6abdf6bc`, where it still has the old directory name. I have tried
- Pointing the github sources to a more recent commit of this repo, which fails on a transitive dependency. After some changes to `cabal.project` it eventually fails because `plutus-tx-plugin` depends on a newer version of ghc than 8.8.3. I had no luck updating `shell.nix` to use a newer ghc.

- Updating `example.cabal` to depend on `language-plutus-core` instead of `plutus-core`, but got some compile errors in `language-plutus-core`, probably also due to a mismatch in GHC versions/libraries

- In the end I just checked out an old commit of the whole repo and /example did build (e300450f61f7e77cf0455171dff4f41290050299)

### PR DESCRIPTION
**Pull request by FinleyMcIlwaine**
This PR fixes #2636 by updating the pinned Plutus library commits to the most recent version and fixing a template-haskell dependency issue. Also updated the development image to a GHC 8.10.2 version. Building the VSCode devcontainer configuration locally with these fixes revealed no problems!

Pinging @j-mueller for review since he reviewed my previous example project update 😃  Thank you!

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `$(nix-build default.nix -A plutus.updateMaterialized)` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `$(nix-shell shell.nix --run fix-stylish-haskell)` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `$(nix-shell shell.nix --run fix-purty)` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [x] Commits have useful messages
- [ ] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

**Comment by FinleyMcIlwaine** [comment]
Added another commit that fixes the `shell.nix` configuration for these updates.

File: example/cabal.project [hunk 1]
```diff
@@ -1,59 +1,61 @@
...
+
 packages:
     ./
 
 source-repository-package
     type: git
     location: https://github.com/input-output-hk/plutus
-    subdir: language-plutus-core
-    tag: 73234550dc2b2c6209975789845bb5ef6abdf6bc
+    subdir: plutus-core
```
Comment (j-mueller) [inline example/cabal.project:9]: you can actually have more than one `subdir` per `source-repository-package`

Comment (FinleyMcIlwaine) [reply example/cabal.project]: Oh, that's nice! I've updated it now so they're all in a list for the plutus ones.

**Comment by j-mueller** [comment]
Thank you! Generally looks good, I'd like to try it out before merging it, I think I can do it this afternoon.

**Review by j-mueller — APPROVED** [review]
Let's merge this. After that I'd like to move the `example` project from this repository to https://github.com/input-output-hk/plutus-starter. That way you can just use it directly without copying it out of the plutus main repo. What do you think @FinleyMcIlwaine ?

**Comment by FinleyMcIlwaine** [comment]
@j-mueller That sounds great, actually! I can move it over there. I can also make the example a little more "full-fledged" so it actually demonstrates basic use of the libraries.

**Comment by j-mueller** [comment]
That would be great, thank you!

**Pull request merged**